### PR TITLE
feat: Add CPM support with envsubst command and development tooling

### PR DESCRIPTION
**Pull request by matt-dresden-caylent**
## Summary

Adds CPM (Caylent Project Manager) support to git-repo with environment variable substitution in manifests, comprehensive development tooling, and test infrastructure improvements.

### 🆕 envsubst Command
- New `repo envsubst` subcommand for environment variable substitution in XML manifests
- Supports `${ENVVAR}` syntax in manifest files
- Recursively processes all XML files in `.repo/manifests/**/*.xml`
- Creates `.bak` backup files before modification
- Comprehensive test coverage (237 test cases)

### 🔧 Repository Configuration
- Updated REPO_URL to `https://github.com/caylent-solutions/git-repo`
- Updated BUG_URL to GitHub issues
- REPO_REV now lazy-loads latest `caylent-*` tag from GitHub API
- Added `repo` script to setup.py for proper installation

### 🛠️ Development Environment
- **Devcontainer**: Full VSCode devcontainer setup with all dependencies
- **Makefile**: Common tasks (test, lint, format, coverage, pre-commit)
- **Git Hooks**: Pre-push validation runs format, lint, and tests
- **.tool-versions**: Version pinning for pre-commit (4.0.1) and python (3.12.9)
- **requirements-dev.txt**: Development dependencies (tox, pytest, black, flake8, isort)

### 🧹 Cleanup
- Removed all man pages (27 files, 3,282 lines deleted)
- Removed manpage generation scripts
- Removed Python 3.8 test runner (Python 3.12 only)

### 🐛 Bug Fixes
- Fixed file copy error handling in project.py (OSError → IOError)
- Removed unnecessary chmod operations on copied files
- Added REPO_REV environment variable to tests to prevent network calls during test execution

### ✅ Testing
- All 287 tests passing
- Added envsubst test suite with edge case coverage
- Simplified tox.ini to Python 3.12 only
- Tests now run without network dependencies

## Files Changed
- **Added** (10 files): devcontainer config, Makefile, git hooks, envsubst command/tests, requirements-dev.txt
- **Modified** (11 files): repo, project.py, pyproject.toml, tox.ini, run_tests, setup.py, .gitignore, .flake8, .isort.cfg, .tool-versions
- **Deleted** (30 files): All man pages, manpage generation scripts, .coverage

## Testing
```bash
make test        # Run all tests
make pre-commit  # Run format, lint, unit-test
```

All tests pass ✅

### CODE REVIEW

File: repo [hunk 1]
```diff
@@ -118,16 +118,36 @@ if __name__ == "__main__":
     check_python_version()
 
 
+def _GetLatestCaylentTag():
```
[the remark this answers is not in the document]
Comment (matt-dresden-caylent) [reply repo]: The warning is correct but not a bug. The check_python_version() function has implicit returns (returns None) in some paths and explicit sys.exit(1) in others. This is intentional - the function either succeeds silently (implicit return) or exits the program. This is fine and doesn't need fixing.
